src/tools/id-mapping/components/results/FromColumn.tsx

```diff
@@ -0,0 +1,45 @@
+import { createContext, useContext } from 'react';
+import { Link } from 'react-router-dom';
+
+import AccessionView from '../../../../shared/components/results/AccessionView';
+
+import { reUniProtKBAccession } from '../../../../uniprotkb/utils';
+
+import { Namespace } from '../../../../shared/types/namespaces';
+import { MappingFlat } from '../../types/idMappingSearchResults';
+import { LocationToPath, Location } from '../../../../app/config/urls';
+
+export const IDMappingFromContext = createContext<Namespace | undefined>(
+  undefined
+);
+
+const From = ({ from }: Partial<MappingFlat>) => {
+  const fromNamespace = useContext(IDMappingFromContext);
+
+  if (!(from && fromNamespace) || fromNamespace === Namespace.idmapping) {
+    return <>{from}</> || null;
+  }
+
+  if (
+    // Not UniProtKB
+    fromNamespace !== Namespace.uniprotkb ||
+    // Or, UniProtKB + "from" is an accession
+    reUniProtKBAccession.test(from)
+  ) {
+    return <AccessionView id={from} namespace={fromNamespace} />;
+  }
+
+  // if it's a UniProtKB ID, return a search with the "direct" flag
```

Comment:
Not sure when we would get here? When running your branch it seems to link directly to the uniprotkb entry

Reply:
It's when you look for an ID, like "A4_HUMAN", and in theory it should redirect you to the entry page, but quickly passing through the search result logic